components/playlist/browser/resources/reducers/player_reducer.ts

```diff
@@ -5,34 +5,108 @@
 
 import { Reducer } from 'redux'
 
+import { PlaylistItem } from 'gen/brave/components/playlist/common/mojom/playlist.mojom.m'
+
 import { PlayerState } from '../reducers/states'
 import { types } from '../constants/player_types'
 import { SelectedPlaylistUpdatedPayload } from '../api/playerApi'
 
+function shuffleItems (
+  itemsInOrder: PlaylistItem[],
+  currentItem: PlaylistItem | undefined
+) {
+  const shuffledItems: PlaylistItem[] = []
+  while (itemsInOrder.length > (currentItem ? 1 : 0)) {
```

Comment:
Maybe we should insert the currentItem at the start, then we wouldn't need to worry about accidentally selecting it in the while loop?

```ts
// If we're playing something, put it at the start of the shuffled items
const shuffledItems = currentItem ? itemsInOrder.splice(0, 1) : []

while (itemsInOrder.length) {
    shuffledItems.push(...itemsInOrder.splice(Math.floor(Math.random() * itemsInOrder.length)))
}

return shuffledItems


```